### PR TITLE
removed case sensitive requirement for commands...

### PR DESCRIPTION
**Pull request by joshtrivedi**
... added toLowerCase() to command triggers, and voice on demand

### CODE REVIEW

File: src/event-distribution/events/message.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const extractMessageInfo: ExtractInfoForEventFunction<
   DiscordEvent.MESSAGE
 > = (message) => {
   const split = message.content.split(" ");
```
Comment (hasanlatch007) [inline src/event-distribution/events/message.ts:44]:
```suggestion
  const split = message.content.toLowerCase().split(" ");
  const trigger = split[0];
  const subTrigger = split[1];
```
The way you've done is basically having it 50/50, any command send the trigger is lowerCased but not the subtrigger, for example `!group Join` would not work.

**Review by hasanlatch007 — CHANGES_REQUESTED** [review]
This actually breaks some of our commands, for example `!timeoutTimer`, it is a trigger with one capitalized letter, with the `toLowerCase()` in [src/event-distribution/events/message.ts](https://github.com/Yes-Theory-Fam/yesbot-ts/pull/667/files#diff-77b198c2b3d38af921c099137af5327d91540a4b70eac40f4391ab96d0de43d8) every message is extracted in lower case and won't be picked up by the new event handler system.

**Review by joshtrivedi — COMMENTED** [review]
reviewed, please verify

File: src/event-distribution/events/message.ts [hunk 2]
```diff
@@ -41,9 +41,9 @@ export const addMessageHandler: AddEventHandlerFunction<
 export const extractMessageInfo: ExtractInfoForEventFunction<
   DiscordEvent.MESSAGE
 > = (message) => {
-  const split = message.content.split(" ");
-  const trigger = split[0];
-  const subTrigger = split[1];
+  const split = message.content.toLowerCase().split(" ");
+  const trigger = split[0] ?? "";
+  const subTrigger = split[1] ?? "";
```
Comment (hasanlatch007) [inline src/event-distribution/events/message.ts:46]: ` ?? ""` Is useless here.

**Review by hasanlatch007 — CHANGES_REQUESTED** [review]
This still breaks commands that have a capital letter like `!timeoutTimer` I would recommend to add this in [src/event-distribution/events/message.ts](https://github.com/Yes-Theory-Fam/yesbot-ts/pull/667/files#diff-77b198c2b3d38af921c099137af5327d91540a4b70eac40f4391ab96d0de43d8) at line `32` and `33`

```suggestion
  const trigger = options.trigger?.toLowerCase() ?? "";
  const subTrigger = options.subTrigger?.toLowerCase() ?? "";
```

File: src/event-distribution/events/message.ts (hunk 2 above)
Comment (joshtrivedi) [reply src/event-distribution/events/message.ts:46]: that was in case there are no sub triggers, it was still passing null

**Review by joshtrivedi — COMMENTED** [review]
check again shawty

**Comment by joshtrivedi** [comment]
something fails help

**Comment by geisterfurz007** [comment]
> something fails help

The testsuite creates snapshot files (basically taking the output, throwing it in a file and saving that) for various outputs of the event distribution. On subsequent testruns, the actual outputs are compared against the file and if the content doesn't match the new result, the test fails.  

Since the whole thing is now case insensitive, the output has changed (the triggers extracted from a message are now converted to lowercase, altering the HandlerInfo returned for some messages in the tests). You can update the tests using `yarn test -u` telling jest that the changes were intentional causing those snapshot files to be written again. Commit those and push them, then the pipeline should go green again 👍

**Review by geisterfurz007 — CHANGES_REQUESTED** [review]
During testing one interaction turned out to still be case-sensitive; the group ping isn't handled by a trigger but a general check if the message.content contains `@group` which currently is not case-insensitive.

A `.toLowerCase()` near the top of group-ping.ts should fix this.

After that, LGTM!

File: src/event-distribution/events/message.ts (hunk 2 above)
Comment (hasanlatch007) [reply src/event-distribution/events/message.ts:46]: I don't know if I would need it for when strictNullsCheck is enabled so Ill keep it.

**Review by geisterfurz007 — APPROVED** [review]
LGTM!

**Review by hasanlatch007 — APPROVED** [review]
Also LGTM!

**Pull request merged**